linux/src/async_socket_linux.c

```diff
@@ -406,12 +411,22 @@ static int initialize_global_thread(void)
                 result = -1;
                 (void)close(g_epoll);
                 g_epoll = -1;
-                interlocked_decrement(&g_thread_access_cnt);
+                (void)interlocked_decrement(&g_thread_access_cnt);
             }
         }
+        if (interlocked_decrement(&g_epoll_access) == 0)
+        {
+            wake_by_address_single(&g_epoll_access);
+        }
```

Comment:
> } [](http://example.com/codeflow?start=7&length=2)

Wouldn't the else branch of this if be horribly and utterly bad?
It would mean someone has incremented the g_epoll_access while we're initializing? That's just pure evil